client/src/App.js

```diff
@@ -14,17 +13,40 @@ export default class App extends Component {
   }
 
   addToSavedList = movie => {
+    console.log("savedList:", this.state.savedList);
+    console.log("movie:", movie);
+    console.log("state check", this.state.savedList.includes(movie));
     const savedList = this.state.savedList;
-    savedList.push(movie);
-    this.setState({ savedList });
+    console.log("var check", savedList.includes(movie));
+    // Added logic so the user cannot save a movie twice:
+    if (savedList.includes(movie)) {
+      alert(
+        `No matter how much you like ${movie.title}, you can only save it once!`
+      );
+      return;
+    } else if (!savedList.includes(movie)) {
+      savedList.push(movie);
+      this.setState({ savedList });
+    }
   };
 
   render() {
     return (
       <div>
-        <SavedList list={this.state.savedList} />
+        <Route
+          render={routeProps => (
+            <SavedList {...routeProps} list={this.state.savedList} />
+          )}
+        />
+        {/* I might not have done this according to best practices, but I figured out how to redirect "/" to "/movielist" automatically: */}
+        <Route exact path="/" component={() => <Redirect to="/movielist" />} />
```

Comment:
I'm glad you figured out Redirects just fine, but you could have bypassed this entirely by having the MovieList render at "/" in the first place! It's not necessarily a best practice thing since you take care of the default path redirecting where you want it to anyway, but most commonly you will see a component rendered at "/" even if it's not a Home page per se